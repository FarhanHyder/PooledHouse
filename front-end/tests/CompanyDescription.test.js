import React from 'react';
import renderer from 'react-test-renderer';
import Description from '../src/Components/ViewTipInfo/CompanyDescription';

test('Description should render correct for single element', () => {
  const thisProps = {'Soho': {
    'tipsPerHour': 12.8888
  }};

  const component = renderer.create(
    <Description locations={thisProps} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('Description should render correct for multiple elements', () => {
  const thisProps = {'Soho': {
    'tipsPerHour': 12.8888
  },
  'Chelsea': {
    'tipsPerHour': 13.54444
  }};

  const component = renderer.create(
    <Description locations={thisProps} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
