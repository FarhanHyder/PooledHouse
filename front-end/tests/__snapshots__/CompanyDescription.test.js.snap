// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correct for multiple elements 1`] = `
<div
  className="container card"
>
  <div
    className="table-responsive-sm"
  >
    <table
      className="table table-dark table-striped table-bordered table-hover"
    >
      <thead
        className="bg-info"
      >
        <tr>
          <th>
            #
          </th>
          <th>
            Company Locations
          </th>
          <th>
            Average Reported Tips
          </th>
          <th>
            Tips in This Zip Code area
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            282 Bowery, New York, NY 10012
          </td>
          <td>
            $
            12.89
             / Hour
          </td>
          <td>
            $
            12.89
             / Hour
          </td>
        </tr>
      </tbody>
      <tbody>
        <tr>
          <td>
            2
          </td>
          <td>
            80 Spring Street, New York, NY 10012
          </td>
          <td>
            $
            13.89
             / Hour
          </td>
          <td>
            $
            12.89
             / Hour
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Description should render correct for single element 1`] = `
<div
  className="container card"
>
  <div
    className="table-responsive-sm"
  >
    <table
      className="table table-dark table-striped table-bordered table-hover"
    >
      <thead
        className="bg-info"
      >
        <tr>
          <th>
            #
          </th>
          <th>
            Company Locations
          </th>
          <th>
            Average Reported Tips
          </th>
          <th>
            Tips in This Zip Code area
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            1
          </td>
          <td>
            282 Bowery, New York, NY 10012
          </td>
          <td>
            $
            12.89
             / Hour
          </td>
          <td>
            $
            12.89
             / Hour
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
